treeherder/etl/tasks/pulse_tasks.py

```diff
@@ -16,7 +17,10 @@ def store_pulse_jobs(job_list, exchange, routing_key):
     newrelic.agent.add_custom_parameter("exchange", exchange)
     newrelic.agent.add_custom_parameter("routing_key", routing_key)
 
-    JobLoader().process_job_list(job_list)
+    try:
+        JobLoader().process_job_list(job_list)
+    except MissingResultsetException as exc:
+        store_pulse_jobs.retry(exc=exc)
```

Comment:
Hmm so I've been thinking about this some more. A plain `.retry()` won't use the retry/jitter from the `retryable_task` I think? And as such will use the defaults of "wait 3 minutes" and will also retry indefinitely:
http://docs.celeryproject.org/en/latest/userguide/tasks.html#retrying

What behaviour is appropriate here? I think perhaps we should stop retrying at some point, at least so invalid data submissions don't get stuck.

Another option might be to modify `retryable_task` such that the try-except has another case for `MissingResultsetException`, where we don't use the `.record_exception()` thus avoiding the New Relic noise, but still keeping the rest of the `retryable_task` benefits.